[Free Trial] Adds Upgrades View

Ecarrion opened the pull request:
closes #9057 

# Why

This PR just adds the main view for the Upgrades section. 

# Testing Steps

It is not connected and has no logic on it yet so just take a look at the code.

# Screenshots

Light | Dark 
--- | ---
<img width="324" alt="light" src="https://user-images.githubusercontent.com/562080/226510772-e3528b4b-900d-4225-8f7b-4093806e711c.png"> |  <img width="331" alt="dark" src="https://user-images.githubusercontent.com/562080/226510778-72bfb42c-852b-4b68-9316-ad5a0875e157.png">

Landscape | Big Fonts | Big Fonts (Scrolled)
--- | --- | ---
<img width="670" alt="landscape" src="https://user-images.githubusercontent.com/562080/226510783-f4943c0c-77fa-45e7-b476-465908e96c99.png"> | <img width="319" alt="big-fonts-1" src="https://user-images.githubusercontent.com/562080/226510780-624ecfff-5e71-480c-a2f8-031bbc14bcce.png"> | <img width="326" alt="big-fonts-2" src="https://user-images.githubusercontent.com/562080/226510784-582a1f50-f000-408d-9ae5-d9ee4bb6c357.png">

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Ecarrion committed:
Adds Upgrades View

ThomazFB reviewed (approved):
Hey @Ecarrion, thanks for introducing this view so quickly! I tested it through the SwiftUI preview and compared it with the Android version. The only thing I noticed different is that we still don't have the `Cancel Free Trial` button there yet. Not sure if that would be a problem, though.

By the way, now that this view is available, this could be a good opportunity for me to jump in with the #9063 and #9064 tasks. Since I already implemented the same thing on the Android side, I can replicate the same work here. , WDYT?

Ecarrion commented:
> The only thing I noticed different is that we still don't have the Cancel Free Trial button there yet. Not sure if that would be a problem, though.

Yes, I added it to be ready for the future but it will be hidden in https://github.com/woocommerce/woocommerce-ios/pull/9230

> By the way, now that this view is available, this could be a good opportunity for me to jump in with the #9063 and #9064 tasks. Since I already implemented the same thing on the Android side, I can replicate the same work here. , WDYT?

Sure, once https://github.com/woocommerce/woocommerce-ios/pull/9230 is merged you'll be in a good position to tackle those tasks!
